docs/100_OVERVIEW.md

```diff
@@ -0,0 +1,147 @@
+# OCR2 Automation Design Overview
+The intent of the plugin design is to separate responsibility of components,
+ensure modularity, and provide accessible types for viewing content in 
+messages via external sources (like telemetry).
+
+## Glossary of Terms
+- Observation: two different types; one from the `Plugin` and one from an `Observer`
+- Report: encoded instructions and/or values for on-chain execution
+- BlockKey: a block identifier (typically a big.Int on many chains)
+- UpkeepIdentifier: an identifier for an upkeep (typically a big.Int)
+- UpkeepKey: a combination of `BlockKey` and `UpkeepIdentifier`
+- NetworkConfig: also called off-chain config in OCR; network configuration
+- In-flight: upkeeps are in-flight if they are reported, but not transmitted
+- Transmit: an attempt to include a report on-chain
+
+## Components
+
+### Plugin
+The `Plugin` provides methods for LibOCR to call as a valid OCR2 plugin. The 
+responsibility of each method is defined below. Generally, a `Plugin` gets
+observations from a list of multiple `Observer`, uses a `Merger` to merge
+observations from the network, and an `Encoder` to produce a transmittable
+report.
+
+#### Rules
+- A `Plugin` only interacts with `Observer`, `Merger`, and `Encoder`
+- A `Plugin` should not know anything about the structural contents of a `BlockKey`
+- A `Plugin` should not know anything about the structural contents of a `UpkeepIdentifier`
+- A `Plugin` should start and stop services per its appropriate life cycle
+- A `Plugin` should treat all incoming observations as potentially malicious
+- A `Plugin` should isolate all state interactions to instances of `Observer`
+
+#### Observation
+The `Observation` method should loop through multiple `Observer` instances to
+collect pending and eligible upkeep results. Those results should be packaged 
+together as an automation observation (distinct type) and byte encoded. This 
+function should never directly interact with the encoded upkeep results. Its 
+sole responsibilities should remain as a collector and basic encoder to the 
+final transport type.
+
+More about specifics of an observation from an observer here:
+[OBSERVATION](./300_OBSERVATION.md)
+
+##### Observation Struct
+An observation is composed of a list of eligible upkeeps and a list of pending
+upkeeps. Providing both allows for all nodes to produce a report without
+checking internal state. Versioning allows for version checks to be applied as
+new nodes can join the network with different active versions.
```

Comment:
do you see there could be more than 1 version of observations in a DON? maybe it's like we release a new version core node which has a new ocr2 keeper plugin. and external Node Ops upgrade at different times?

i feel this will make Report function and any other observation bytes decoding process more difficult.